src/libs/konnectors/stopKonnectorJob.ts

```diff
@@ -0,0 +1,61 @@
+import Minilog from '@cozy/minilog'
+
+import CozyClient, { Q } from 'cozy-client'
+
+import { TIMEOUT_KONNECTOR_ERROR } from '/libs/Launcher'
+
+const log = Minilog('stopKonnectorJob')
+
+export const stopKonnectorJob = async (
+  client: CozyClient,
+  jobId: string
+): Promise<void> => {
+  try {
+    await stopJob(client, jobId)
+    await findJobStopTriggerAndRemove(client, jobId)
+  } catch (e) {
+    log.error('Error while trying to stop job ${jobId}')
+    log.error(e)
+  }
+}
+
+/**
+ * Mark the given job as "context deadline exceeded"
+ */
+const stopJob = (client: CozyClient, jobId: string): Promise<unknown> => {
+  return client.save({
```

Comment:
Don't we need to check if the job is still running on cozy-stack first? What happens if we do this after the stack run the cleanup service?

Reply:
We can. I was trying to avoid to get the job. But since I will need to get the job to pass it's real revision, I will be able to check this too.

If we do this after the cleanup service of the stack, the last one to update the job will win.